include/envoy/server/listener_manager.h

```diff
@@ -131,6 +131,16 @@ class ListenerManager {
     All,
   };
 
+  // the types of listeners to be returned from listeners(ListenerState)
+  // an enum instead of enum class so the underlying type is an int and bitwise operations can be
+  // used without casting
+  enum ListenerState : uint8_t {
+    ACTIVE = 1 << 0,
+    WARMING = 1 << 1,
+    DRAINING = 1 << 2,
+    ALL = 1 << 3
```

Comment:
The advantage of the use of shifted bits is that you can do this:

  ALL = ACTIVE | WARMING | DRAINING

Reply:
Ah right yes, I'll change this cheers